v2/cmux.go

```diff
@@ -73,3 +67,16 @@ type timeoutError struct{}
 func (e *timeoutError) Error() string   { return "i/o timeout" }
 func (e *timeoutError) Timeout() bool   { return true }
 func (e *timeoutError) Temporary() bool { return true }
+
+type Session interface {
+	OpenStream() (net.Conn, error)
+	AcceptStream() (net.Conn, error)
+	Close() error
+	NumStreams() int
+}
+
+type Protocol interface {
+	Client(net.Conn, *DialerOpts) (Session, error)
+	Server(net.Conn, *ListenOpts) (Session, error)
```

Comment:
Are `DialerOpts` and `ListenOpts` reserved for future use? They are not used for now. Are the protocol implementations supposed to understand cmux configurations?

Reply:
true, could cut these out to simplify, I don't think any of the options we have will mean anything to any of the protocols.